cli/bpmetadata/tfconfig.go

```diff
@@ -279,10 +310,30 @@ func getBlueprintRequirements(rolesConfigPath, servicesConfigPath string) (*Blue
 		return nil, err
 	}
 
-	return &BlueprintRequirements{
-		Roles:    r,
-		Services: s,
-	}, nil
+	if versionsConfigPath != "" {
+		//parse blueprint provider versions
+		versionsFile, diags := p.ParseHCLFile(versionsConfigPath)
+		err = hasHclErrors(diags)
+		if err != nil {
+			return nil, err
+		}
+
+		v, err := parseBlueprintProviderVersions(versionsFile)
+		if err != nil {
+			return nil, err
+		}
+
+		return &BlueprintRequirements{
+			Roles:            r,
+			Services:         s,
+			ProviderVersions: v,
+		}, nil
+	} else {
```

Comment:
nit: prefer to flip this logic to keep left aligned and can remove else block

```
if file does not exist {
return &BlueprintRequirements{
			Roles:    r,
			Services: s,
		}, nil
}

versionsFile, diags := p.ParseHCLFile(versionsConfigPath)
err = hasHclErrors(diags)
....

```

Reply:
Done